code/modules/antagonists/heretic/status_effects/buffs.dm

```diff
@@ -22,15 +27,29 @@
 	owner.pass_flags &= ~(PASSCLOSEDTURF | PASSGLASS | PASSGRILLE | PASSMACHINE | PASSSTRUCTURE | PASSTABLE | PASSMOB | PASSDOORS | PASSVEHICLE)
 	owner.forceMove(location)
 	location = null
+	cancel_button.Remove(owner)
 
 /datum/status_effect/crucible_soul/get_examine_text()
 	return span_notice("[owner.p_They()] [owner.p_do()]n't seem to be all here.")
 
+/datum/action/cancel_crucible_soul
+	name = "Recall"
+	desc = "Use to end the blessing early"
+	button_icon = 'icons/obj/antags/eldritch.dmi'
+	button_icon_state = "crucible_soul"
+
+/datum/action/cancel_crucible_soul/Trigger(trigger_flags)
+	. = ..()
+	if(!.)
+		return
+	var/datum/status_effect/active_effect = owner.has_status_effect(/datum/status_effect/crucible_soul)
+	active_effect.Destroy()
+
 // DUSK AND DAWN
 /datum/status_effect/duskndawn
 	id = "Blessing of Dusk and Dawn"
 	status_type = STATUS_EFFECT_REFRESH
-	duration = 60 SECONDS
+	duration = 3 MINUTES
```

Comment:
Also thinking about it now, this change + cheaper potions make me a little worry it's too easy to get 100% uptime on this.

Reply:
Made it 90 seconds, shouldn't guarantee 100% uptime but grant a little extra leeway.